inc/datahandlers/pm.php

```diff
@@ -293,19 +293,22 @@ function verify_recipient()
 			// - sender is an administrator
 			if(($this->admin_override != true && $sender_permissions['cancp'] != 1) && $sender_permissions['canoverridepm'] != 1)
 			{
-				$ignorelist = explode(",", $user['ignorelist']);
-				if(!empty($ignorelist) && in_array($pm['fromid'], $ignorelist))
+				if(!empty($user['ignorelist']))
 				{
-					$this->set_error("recipient_is_ignoring", array($user['username']));
+					$ignorelist = explode(',', $user['ignorelist']);
+					if(!empty($ignorelist) && in_array($pm['fromid'], $ignorelist))
```

Comment:
`!empty()` not needed here and in line 309, they don't do anything because `$user['ignorelist']` and `$user['buddylist']` are checked earlier and they're surely filled there.